app/src/main/kotlin/com/simplemobiletools/calculator/views/SketchSheetView.java

```diff
@@ -0,0 +1,133 @@
+package com.simplemobiletools.calculator.views;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.view.MotionEvent;
+import android.view.View;
+
+import org.jetbrains.annotations.TestOnly;
+
+import java.util.ArrayList;
+
+public class SketchSheetView extends View {
+
+    private Paint paint;
+    private Path path;
+    private Canvas canvas;
+    private ArrayList<ArrayList<int[]>> points;
+
+    public SketchSheetView(Context context) {
+        super(context);
+        this.setBackgroundColor(Color.WHITE);
+        canvas = new Canvas(Bitmap.createBitmap(820, 480, Bitmap.Config.ARGB_4444));
+        points = new ArrayList<>();
+        path = new Path();
+        paint = new Paint();
+        paint.setDither(true);
+        paint.setColor(Color.parseColor("#000000"));
+        paint.setStyle(Paint.Style.STROKE);
+        paint.setStrokeJoin(Paint.Join.ROUND);
+        paint.setStrokeCap(Paint.Cap.ROUND);
+        paint.setStrokeWidth(4);
+    }
+
+    private ArrayList<DrawingClass> DrawingClassArrayList = new ArrayList<>();
+
+    @Override
+    public boolean onTouchEvent(MotionEvent event) {
+        DrawingClass pathWithPaint = new DrawingClass();
+        pathWithPaint.setPaint(paint);
+        canvas.drawPath(path, paint);
+
+        if (event.getAction() == MotionEvent.ACTION_DOWN) {
+            startNewStroke();
+            addPoint(Math.round(event.getX()), Math.round(event.getY()));
+            path.moveTo(event.getX(), event.getY());
+            path.lineTo(event.getX()-1, event.getY()-1);
+            path.lineTo(event.getX(), event.getY());
+            pathWithPaint.setPath(path);
+            DrawingClassArrayList.add(pathWithPaint);
+        }
+        else if (event.getAction() == MotionEvent.ACTION_MOVE) {
+            addPoint(Math.round(event.getX()), Math.round(event.getY()));
+            path.lineTo(event.getX(), event.getY());
+            pathWithPaint.setPath(path);
+            DrawingClassArrayList.add(pathWithPaint);
```

Comment:
If I understand correctly, only the last added element in DrawingClassArrayList is ever being used by onDraw(). Seems it would be safe to use a single DrawingClass variable instead of an array list, and overwrite it on every touch event.

Reply:
Gonna look into that

Reply:
fixed